runtime/test/src/test.rs

```diff
@@ -1389,3 +1390,143 @@ async fn test_store_set_id() {
+            let host_exports = host_exports::test_support::HostExports::new(&ctx);
+
+            let metrics_registry = Arc::new(MetricsRegistry::mock());
+            let stopwatch = StopwatchMetrics::new(
+                ctx.logger.clone(),
+                deployment.hash.clone(),
+                "test",
+                metrics_registry.clone(),
+            );
+            let gas = GasCounter::new();
+
+            Host {
+                ctx,
+                host_exports,
+                stopwatch,
+                gas,
+            }
+        }
+
+        fn store_set(
+            &mut self,
+            entity_type: &str,
+            id: &str,
+            data: Vec<(&str, &str)>,
+        ) -> Result<(), HostExportError> {
+            let data: Vec<_> = data.into_iter().map(|(k, v)| (k, Value::from(v))).collect();
+            self.store_setv(entity_type, id, data)
+        }
+
+        fn store_setv(
+            &mut self,
+            entity_type: &str,
+            id: &str,
+            data: Vec<(&str, Value)>,
+        ) -> Result<(), HostExportError> {
+            let id = String::from(id);
+            let data = HashMap::from_iter(data.into_iter().map(|(k, v)| (Word::from(k), v)));
+            self.host_exports.store_set(
+                &self.ctx.logger,
+                &mut self.ctx.state,
+                &self.ctx.proof_of_indexing,
+                entity_type.to_string(),
+                id,
+                data,
+                &self.stopwatch,
+                &self.gas,
+            )
+        }
+    }
+
+    #[track_caller]
+    fn err_says<E: std::fmt::Debug + std::fmt::Display>(err: E, exp: &str) {
+        let err = err.to_string();
+        assert!(err.contains(exp), "expected `{err}` to contain `{exp}`");
+    }
+
+    const UID: &str = "u1";
+    const USER: &str = "User";
+    const BID: &str = "0xdeadbeef";
+    const BINARY: &str = "Binary";
+
+    let mut host = Host::new().await;
+
+    host.store_set(USER, UID, vec![("id", "u1"), ("name", "user1")])
+        .unwrap();
+
+    let err = host
+        .store_set(
+            USER,
+            UID,
+            vec![
+                ("id", "u1"),
+                ("name", "user1"),
+                ("test", "invalid_field"),
+                ("test2", "invalid_field"),
+            ],
+        )
+        .err()
+        .unwrap();
+
+    // The order of `test` and `test2` is not guranteed
+    // So we just check the string contains them
+    let err_string = err.to_string();
+    dbg!(err_string.as_str());
+    assert!(err_string
+        .contains("The provided entity has fields not defined in the schema for entity `User`"));
+    assert!(err_string.contains("test"));
+    assert!(err_string.contains("test2"));
+
+    let err = host
+        .store_set(
+            USER,
+            UID,
+            vec![("id", "u1"), ("name", "user1"), ("test3", "invalid_field")],
+        )
+        .err()
+        .unwrap();
+
+    err_says(
```

Comment:
Could be worth creating a ValidationError with the current anonymous errors and check on types if you feel up for that

Reply:
Here you should now be able to check the error returned rather than the message error, this is why the enum is useful

Reply:
The problem is its no longer `EntityValidationError` anymore it get anonymised in between an only the message gets passed into `HostExportError:Deterministic`

Reply:
🤦‍♂️ okay, oh well